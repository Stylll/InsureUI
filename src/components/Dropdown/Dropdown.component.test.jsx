import { render, fireEvent } from '@testing-library/react';
import React from 'react';

import Dropdown from './Dropdown.component';

describe('Dropdown Component', () => {
  const categories = [
    { id: '1', name: 'Electronics' },
    { id: '2', name: 'Automobile' },
    { id: '3', name: 'Kitchen' },
  ];
  it('should render properly', () => {
    const { baseElement } = render(<Dropdown
            name="category"
            value="1"
            error=""
            onChange={() => {}}
            options={categories}
            optionsTitleProperty="name"
            optionsValueProperty="id"
        />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should call onChange method when dropdownlist is changed', () => {
    const onChange = jest.fn();
    const { getByTestId } = render(<Dropdown
        name="category"
        value="1"
        error=""
        onChange={onChange}
        options={categories}
        optionsTitleProperty="name"
        optionsValueProperty="id"
        dataTestId="dropdownlist-test"
    />);

    fireEvent.change(getByTestId('dropdownlist-test'), { target: { value: '2' } });
    expect(onChange).toHaveBeenCalled();
  });
});
