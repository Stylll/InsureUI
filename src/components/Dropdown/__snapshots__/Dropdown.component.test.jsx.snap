// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component should render properly 1`] = `
<body>
  <div>
    <div
      class="drop-down"
    >
      <select
        class="select"
        data-testid="dropdown"
        name="category"
      >
        <option
          value=""
        />
        <option
          value="1"
        >
          Electronics
        </option>
        <option
          value="2"
        >
          Automobile
        </option>
        <option
          value="3"
        >
          Kitchen
        </option>
      </select>
      <span
        class="chevy"
      >
        âˆ¨
      </span>
      <span
        class="info"
      />
      
    </div>
  </div>
</body>
`;
